### PR TITLE
feat: Add options to ignore specific urls and regex  patterns

### LINKED ISSUE #68: Changed `ignoredHosts` to `ignoredPatterns`
**Feature Request**
I want to filter out the `/ping` calls from the server done by `NetInfo` but they're in the same host as my main server that I am trying to monitor.

If we say the request has the following text
```
HEAD https://trajano.net/ping
```

It would simplify us by having rules that look like

```
ignoredPatterns: [
  /^HEAD https:\/\/trajano.net\/ping$/
]
```
**Additional context**

### PR DESCRIPTION
**Pull request by alexbrazier**
Closes #68 

Adds new `ignoredUrls` and `ignoredPatterns` to provide full urls to ignore or regex patterns to ignore.

The regex patterns will be slightly slower but gives more control to ignore certain methods or partial urls.

### CODE REVIEW

**Comment by trajano** [comment]
I think I liked your idea of just letting us pass a callback in #68  as it would simplify your code and we can do whatever we need.
such as

* ignore anything that's less than 100ms so we can focus on slow API responses.
* ignore if we pass something in the header.

But those can be done later.

File: src/Logger.ts [hunk 1]
```diff
@@ -152,6 +166,23 @@ export default class Logger {
       this.ignoredHosts = new Set(options.ignoredHosts);
     }
 
+    if (options?.ignoredPatterns) {
```
Comment (trajano) [inline src/Logger.ts:169]:
```
this.ignoredPatterns = options?.ignoredPatterns ?? [];
```

Would shrink the code down

File: src/Logger.ts [hunk 2]
```diff
@@ -152,6 +166,23 @@ export default class Logger {
...
+      this.ignoredPatterns = options.ignoredPatterns;
+    }
+
+    if (options?.ignoredUrls) {
+      if (
+        !Array.isArray(options.ignoredUrls) ||
+        typeof options.ignoredUrls[0] !== 'string'
+      ) {
+        warn(
```
Comment (trajano) [inline src/Logger.ts:178]: this may be better as a `throw Error()` rather than a warning.  This will prompt the developer to fix their code.

Comment (alexbrazier) [reply src/Logger.ts:178]: Yeah, I can't remember why I did the warning as it was a while ago when I set it up. I would probably accept a PR updating it.

**Comment by alexbrazier** [comment]
Released in `1.14.0`

**Pull request merged**